core/templates/dev/head/components/HintAndSolutionButtonsDirective.js

```diff
@@ -57,7 +56,8 @@ oppia.directive('hintAndSolutionButtons', [
           $scope.isHintButtonVisible = function(index) {
             return (
               HintsAndSolutionManagerService.isHintViewable(index) &&
-              ExplorationEngineService.doesInteractionSupportHints());
+              ExplorationPlayerStateService.getCurrentEngineService().
```

Comment:
This really should be ExplorationPlayerStateService.doesInteractionSupportHints(), and ditto below. All the "current engine" stuff can be kept within the exploration player state service.

OK to update now or in a future PR.

Reply:
Done, fixed it here itself.